OnProfileUpdate/__tests__/handler.test.ts

```diff
@@ -0,0 +1,210 @@
+    email: "cittadinanzadigitale@teamdigitale.governo.it",
+    fiscalCode: "ISPXNB32R82Y766D",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    acceptedTosVersion: 1,
+    id: "ISPXNB32R82Y766D-0000000000000001",
+    version: 1,
+    _self: "eb28139a-f875-4276-81c5-b3f7b01f712a"
+  },
+  {
+    fiscalCode: "not-valid",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    acceptedTosVersion: 1,
+    id: "id",
+    version: 0
+  },
+  {
+    email: "Eleanore.Kuphal@example.net",
+    fiscalCode: "VSFNVG14A39Y596X",
+    isEmailValidated: false,
+    isInboxEnabled: false,
+    id: "VSFNVG14A39Y596X-0000000000000001",
+    version: 1,
+    acceptedTosVersion: 0,
+    _self: "aa406556-b003-4387-89ef-8f127ec9b2da"
+  },
+  {
+    email: "Eleanore.Kuphal@example.net",
+    fiscalCode: "VSFNVG14A39Y596X",
+    isEmailValidated: true,
+    isInboxEnabled: false,
+    id: "VSFNVG14A39Y596X-0000000000000002",
+    version: 2,
+    _self: "f43db63b-5549-403d-98a5-e781934c796f"
+  }
+];
+
+jest.spyOn(mockProfileModel, "find").mockImplementation(([id, fiscalCode]) =>
+  pipe(
+    mockProfiles.find(
+      profile => profile.id === id && profile.fiscalCode === fiscalCode
+    ),
+    foundProfile => TE.right(O.fromNullable(foundProfile as RetrievedProfile))
+  )
+);
+
+const mockLogger = {
+  error: jest.fn()
+};
+
+const mockDependencies = {
+  dataTableProfileEmailsRepository: mockDataTableProfileEmailsRepository,
+  profileModel: mockProfileModel,
+  logger: mockLogger
+};
+
+describe("handler function", () => {
+  it("should call get, insert and delete methods with no errors", async () => {
+    const documents = mockProfiles;
+
+    await handler(documents)(mockDependencies)();
+
+    expect(mockProfileModel.find).toHaveBeenCalledTimes(5);
+    expect(mockDataTableProfileEmailsRepository.insert).toHaveBeenCalledTimes(
+      5
+    );
+    expect(mockDataTableProfileEmailsRepository.delete).toHaveBeenCalledTimes(
+      1
+    );
+  });
+
+  it("should call mockLogger.error when error in insertProfileEmail occurs", async () => {
+    const mockDocuments = mockProfiles.filter(
+      mockProfile => mockProfile.id === "DRUQIL23A18Y188X-0000000000000000"
+    );
+
+    jest
+      .spyOn(mockDataTableProfileEmailsRepository, "insert")
+      .mockImplementationOnce(() => Promise.reject(new Error("Insert error")));
+
+    await handler(mockDocuments)(mockDependencies)();
+
+    expect(mockLogger.error).toHaveBeenCalledWith(
+      "error handling profile with _self ba130521-8bab-4a68-a5e9-07a7e59f1f24"
+    );
+  });
+
+  it("should call mockLogger.error when error in find occurs", async () => {
+    const mockDocuments = mockProfiles.filter(
+      mockProfile => mockProfile.id === "PVQEBX22A89Y092X-0000000000000001"
+    );
+
+    jest
+      .spyOn(mockProfileModel, "find")
+      .mockImplementationOnce(() =>
+        TE.left({ kind: "COSMOS_CONFLICT_RESPONSE" })
+      );
+
+    await handler(mockDocuments)(mockDependencies)();
```

Comment:
We want to ensure that at lest one E.Left is returned (to throw a retry). Check the result.